python/lsst/meas/algorithms/measureApCorr.py

```diff
@@ -34,32 +35,39 @@
 from .sourceSelector import sourceSelectorRegistry
 
 
-class FluxKeys:
-    """A collection of keys for a given flux measurement algorithm
-    """
-    __slots__ = ("flux", "err", "flag", "used")  # prevent accidentally adding fields
+class _FluxNames:
+    """A collection of flux-related names for a given flux measurement algorithm.
 
+    Parameters
+    ----------
+    name : `str`
+        Name of flux measurement algorithm, e.g. ``base_PsfFlux``.
+    schema : `lsst.afw.table.Schema`
+        Catalog schema containing the flux field.  The ``{name}_instFlux``,
+        ``{name}_instFluxErr``, ``{name}_flag`` fields are checked for
+        existence, and the ``apcorr_{name}_used`` field is added.
+
+    Raises
+    ------
+    KeyError if any of instFlux, instFluxErr, or flag fields is missing.
+    """
     def __init__(self, name, schema):
-        """Construct a FluxKeys
-
-        Parameters
-        ----------
-        name : `str`
-            Name of flux measurement algorithm, e.g. "base_PsfFlux"
-        schema : `lsst.afw.table.Schema`
-            Catalog schema containing the flux field
-                read: {name}_instFlux, {name}_instFluxErr, {name}_flag
-                added: apcorr_{name}_used
-        """
-        self.flux = schema.find(name + "_instFlux").key
-        self.err = schema.find(name + "_instFluxErr").key
-        self.flag = schema.find(name + "_flag").key
-        self.used = schema.addField("apcorr_" + name + "_used", type="Flag",
-                                    doc="set if source was used in measuring aperture correction")
+        self.fluxName = name + "_instFlux"
+        if not schema.extract(self.fluxName):
```

Comment:
`if self.fluxName not in schema:` is more idiomatic, and I sure hope it's not slower.